frog_api/views/data.py

```diff
@@ -96,43 +98,50 @@ def get(self, request: Request, project_slug: str) -> Response:
 def get_progress_shield(
     request: Request, project_slug: str, version_slug: str, category_slug: str
 ) -> dict[str, Any]:
-    latest = get_latest_entry(project_slug, version_slug, "default")
+    latest = get_latest_entry(project_slug, version_slug, category_slug)
+    latest_measures = latest[0]["measures"]
 
     project = get_project(project_slug)
     version = get_version(version_slug, project)
     category = get_category(category_slug, version)
 
     params = request.query_params
     if not params:
-        raise InvalidDataException("No measure or total specified")
+        raise InvalidDataException("No measure specified")
 
-    try:
+    if "measure" in params:
         measure = params["measure"]
-        numerator = latest[0]["measures"][measure]
-    except:
+    else:
         raise InvalidDataException("No measure specified")
+    if measure not in latest_measures:
+        raise InvalidDataException(f"Measure '{measure}' not found")
+    numerator = latest_measures[measure]
 
     label = params.get(
         "label",
         " ".join(
             [
                 version.name,
                 category.name,
-                str(measure),
+                title(str(measure)),
```

Comment:
I'm not sure this should be done. I get preferring 'Total' to 'total" though... Hmm

Reply:
Having separate name/slug for measures could be useful, but that's a big schema change

Reply:
for now we can title-ify it, but yeah I may wanna revisit this.